### PR TITLE
APP-2132 - Add Warning for Minting tokens to DAO treasury

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- warning modal added
- update address validation for mint token action

Task: [APP-2132](https://aragonassociation.atlassian.net/browse/APP-2132)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2132]: https://aragonassociation.atlassian.net/browse/APP-2132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/web-app/src/containers/mintTokensToTreasuryMenu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            <FormTitle>{t('modal.mintTokensToTreasury.inputLabel')}</FormTitle>
+            <FormHelpText>
+              {t('modal.mintTokensToTreasury.inputHelptext')}
+            </FormHelpText>
+            <Controller
+              defaultValue=""
+              name={'mintTokensToTreasury'}
+              control={control}
+              rules={{
+                required: t('errors.required.walletAddress') as string,
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/mintTokensToTreasuryMenu/index.tsx]: Can you remove the casting please?

File: packages/web-app/src/containers/mintTokensToTreasuryMenu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                    name={name}
+                    value={value}
+                    onBlur={onBlur}
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                      onChange(e.target.value);
+                    }}
+                    placeholder="0x"
+                    adornmentText={value ? t('labels.copy') : t('labels.paste')}
+                    onAdornmentClick={() =>
+                      handleClipboardActions(value, onChange, alert)
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/mintTokensToTreasuryMenu/index.tsx:134]: I think you're using the wrong alert here. This is the browser one, but should be the chip alert.

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx [hunk 3]
```diff
@@ -341,6 +354,13 @@ export const MintTokenForm: React.FC<MintTokenFormProps> = ({
     }, 450);
   };
 
+  const openMintoTreasuryModal = (index: number) => {
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: Misspelling?

File: packages/web-app/src/locales/en/common.json [hunk 4]
```diff
@@ -935,6 +935,20 @@
       "titleProposalCreation": "Exit proposal creation",
       "titleDaoCreation": "Exit DAO creation"
     },
+    "mintTokensToTreasury": {
+      "title": "Minting tokens to DAO treasury",
+      "description": "If you are minting tokens for your DAO treasury, you may brick your DAO because you will never again reach minimum participation and thus be unable to execute any proposal actions.",
+      "alertTitle": "Potential risk for your DAO",
+      "alertDescription": "Aragon doesn’t recommend to mint tokens to your DAO treasury. You are always able to make a proposal to distribute new tokens to new members.",
+      "step1CtaLabel": "OK, I have understood the risk",
```
Comment (Fabricevladimir) [inline packages/web-app/src/locales/en/common.json:943]: This is rather counterintuitive and confusing. When I say I understand the risk, I expect it to mean that I understand the risk let me do what I want to do anyway.

File: packages/web-app/src/containers/mintTokensToTreasuryMenu/index.tsx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  WalletInput,
+} from '@aragon/ui-components';
+
+import ModalBottomSheetSwitcher from 'components/modalBottomSheetSwitcher';
+import {StateEmpty} from 'components/stateEmpty';
+import {validateAddress} from 'utils/validators';
+import {handleClipboardActions} from 'utils/library';
+
+const processes = ['DaoCreation', 'ProposalCreation'] as const;
+export type ProcessType = typeof processes[number];
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/mintTokensToTreasuryMenu/index.tsx]: What is the use of these two?

**Review by Fabricevladimir — CHANGES_REQUESTED** [review]
Thanks Sepehr. Looks good. Left a couple minor coments

File: packages/web-app/src/containers/actionBuilder/mintTokens/addressTokenRow.tsx [hunk 6]
```diff
@@ -69,6 +80,14 @@ const AddressField: React.FC<AddressFieldProps> = ({
         }
       });
     }
+
+    if (onEnterDaoAddress && validationResult === true) {
+      // do not open the modal for more than one time for the same address
+      if ((address === daoAddress || address === ensName) && !isModalOpened) {
```
Comment (jamesej) [inline packages/web-app/src/containers/actionBuilder/mintTokens/addressTokenRow.tsx]: Comparison needs to be case-insensitive

**Review by jamesej — CHANGES_REQUESTED** [review]
Also there is an issue where the modal extends off the right hand side in tablet view (but not mobile for some reason). You need to make the browser window maybe 1000px wide but 1600 high or something like that to see this one.

File: packages/web-app/src/locales/en/common.json (hunk 4 above)
Comment (sepehr2github) [reply packages/web-app/src/locales/en/common.json:943]: We need to check this with selim

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx [hunk 7]
```diff
@@ -143,6 +145,17 @@ export const MintTokenForm: React.FC<MintTokenFormProps> = ({
...
   );
   const [newHoldersCount, setNewHoldersCount] = useState(0);
+  const [mintTokensToTreasuryModal, setMintTokensToTreasuryModal] = useState<{
+    status: boolean;
+    index?: number;
+  }>({
+    status: false,
+  });
+
+  const isModalOpened = useMemo(
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: The calculation is one check and the dependency the same, at this point, memoizing might actually be harmful!

**Review by Fabricevladimir — APPROVED** [review]
Thanks Sepehr, looks good. This can be merged once you've addressed Jame's comment about the tablet mode

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx (hunk 7 above)
Comment (jamesej) [reply packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: Fixed this one

**Review by jamesej — APPROVED** [review]
All good now

**Pull request merged**